src/vocgui/views/training_set_by_id_viewset.py

```diff
@@ -0,0 +1,47 @@
+from rest_framework import viewsets
+from django.db.models import Count, Q
+from vocgui.models import TrainingSet
+from vocgui.serializers import TrainingSetSerializer
+from .utils import check_group_object_permissions
+from django.http import Http404
```

Comment:
```suggestion
from django.core.exceptions import PermissionDenied
from django.db.models import Count, Q
from ..models import TrainingSet, GroupAPIKey
from ..serializers import TrainingSetSerializer
from ..utils import get_key
```
If you go with my suggestions, these imports would be required